Code/Tools/AssetProcessor/native/utilities/ApplicationManagerBase.cpp

```diff
@@ -1704,6 +1704,12 @@ void ApplicationManagerBase::RegisterBuilderInformation(const AssetBuilderSDK::A
 
                 do
                 {
+                    if (jobCancelListener.IsCancelled())
+                    {
+                        // do not attempt to continue to retry or spawn
+                        // new builders during shut down.
+                        break;
+                    }
```

Comment:
Should this also be applied to the CreateJobs call above?  Also isn't there a shutdown requested ebus that's different from the cancel request? `AssetBuilderCommandBus::Shutdown`

Reply:
I'll investigate.  I need to retest this anyway - I think flaky tests are causing problems.

Reply:
After looking at AssetBuilderCommandBus::Shutdown its meant for individual asset builders to listen to it (not jobs) so its byId and Single.  I'd rather not change that architecture.

As for the same logic in CreateJobs, I have applied it and will upload a new PR.  Both CreateJobs and ProcessJob use the same retry function, so I shored that up too internally and will make it break out immediately if it can't find a new builder instead of exponential backoff.